src/renderer/api/kube-api.ts

```diff
@@ -370,62 +372,71 @@ export class KubeApi<T extends KubeObject = any> {
     if (!opts.abortController) {
       opts.abortController = new AbortController();
     }
+    let errorReceived = false;
     const { abortController, namespace, callback } = opts;
```

Comment:
Is there a reason why we are using a callback here? Does it make the API easier to use, vs a promise?

Reply:
@Nokel81 I guess it's done to be `KubeApi.watch()` a `sync`-call

Reply:
I went with callback because it's backward compatible (we are returning disposer functions).